libkbfs/config_local.go

```diff
@@ -594,6 +595,18 @@ func (c *ConfigLocal) ResetCaches() {
 		int64(MaxBlockSizeBytesDefault * maxParallelBlockPuts * 2)
 	c.dirtyBcache = NewDirtyBlockCacheStandard(c.clock, c.MakeLogger,
 		minSyncBufferSize, maxSyncBufferSize)
+	if oldDirtyBcache != nil {
+		// Shutdown in the background so we don't block future access
+		// while holding the config write lock.
+		go func() {
```

Comment:
Hmm. Not a huge fan of this, since it shuts down the old cache non-deterministically. If we someday have auto-goroutine-leak detection on test shutdown (which is plausible), then that might fail flakily.

Can you instead just do this outside the lock after everything else is done? This probably requires moving all this code into another helper function.